scipy/_lib/_array_api.py

```diff
@@ -652,3 +657,115 @@ def xp_default_dtype(xp):
+
+    for backend, _ in skip_backends:  # ignoring the reason
+        setattr(capabilities[backend], 'cpu', False)
+        setattr(capabilities[backend], 'gpu', False)
+        setattr(capabilities[backend], 'mask', False)
+
+    if cpu_only:
+        cupy.gpu = False
+        torch.gpu = False
+        jax.gpu = False
+        dask.gpu = False
+
+    return capabilities
+
+
+def _make_capabilities_table(capabilities):
+    numpy = capabilities['numpy']
+    cupy = capabilities['cupy']
+    torch = capabilities['torch']
+    jax = capabilities['jax']
+    dask = capabilities['dask']
+
+    for backend in [numpy, cupy, torch, jax, dask]:
+        for attr in ['cpu', 'gpu', 'mask']:
+            val = "✓" if getattr(backend, attr) else " "
+            val += " " * (len('{numpy.}') + len(attr) - 1)
+            setattr(backend, attr, val)
+
+    table = f"""                
+    +---------+-------------+-------------+--------------+
+    | Library | CPU         | GPU         | ``marray``   |
+    +=========+=============+=============+==============+
+    | NumPy   | {numpy.cpu} | {numpy.gpu} | {numpy.mask} |
+    +---------+-------------+-------------+--------------+
+    | CuPy    | {cupy.cpu } | {cupy.gpu } | {cupy.mask } |
+    +---------+-------------+-------------+--------------+
+    | PyTorch | {torch.cpu} | {torch.gpu} | {torch.mask} |
+    +---------+-------------+-------------+--------------+
+    | JAX     | {jax.cpu  } | {jax.gpu  } | {jax.mask  } |
+    +---------+-------------+-------------+--------------+
+    | Dask    | {dask.cpu } | {dask.gpu } | {dask.mask } |
+    +---------+-------------+-------------+--------------+
+    """
+    return table
+
+
+def _make_capabilities_note(fun_name, capabilities):
+    table = _make_capabilities_table(capabilities)
+    note = f"""
+    `{fun_name}` has experimental support for Python Array API Standard compatible
+    backends in addition to NumPy. Please consider testing these features
+    by setting an environment variable ``SCIPY_ARRAY_API=1`` and providing
+    CuPy, PyTorch, JAX, or Dask arrays as array arguments. The following
+    combinations of backend and device (or other capability) are supported.
+    {table}
+     See :ref:`dev-arrayapi` for more information.
+    """
+    return textwrap.dedent(note)
+
+
+def xp_capabilities(capabilities_table):
+    def decorator(f):
+        fun_name = f.__name__
+        skip_backends = capabilities_table[fun_name].get('skip_backends', None)
+        cpu_only = capabilities_table[fun_name].get('cpu_only', None)
+        capabilities = _make_capabilities(skip_backends=skip_backends,
+                                          cpu_only=cpu_only)
+
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            return f(*args, **kwargs)
+
+        note = _make_capabilities_note(fun_name, capabilities)
+        doc = FunctionDoc(wrapper)
+        doc['Notes'].append(note)
```

Comment:
Does `numpydoc` allow us define a new section like `Array API Compatibility` for better visibility rather than being at the end of the notes?

Reply:
I'm not sure, but if not, we could ask.

Reply:
Not required for this pr though but it would be nice for the future